### DISCUSSION TITLE
Resetting elements in a container

### DISCUSSION THREAD

**Discussion by BillLi1992** [zauberzeug/nicegui#2622] [Q&A]
### Question

Hi, 

Is there a way to reset the elements declared in a container?

For example:

```python
container = ui.row() 
with container:  
   input = ui.input()
```

if I call 'container.clear()', it will delete the input element. Instead I'm looking for someway to reset all the child elements in the container to their initial value.

Thanks.

**Comment by falkoschindler** [comment]
Hi @BillLi1992!

You can use the [`@refreshable`](https://nicegui.io/documentation/refreshable) decorator to declare a piece of UI which can be "refreshed" like this:
```py
@ui.refreshable
def container():
    with ui.row():
        ui.input()

container()

ui.button('Refresh', on_click=container.refresh)
```
This will clear the container and recreate it's content.

**Reply by BillLi1992** [reply]
Hi Falko, 

Thanks for pointing this out. My limitation with ui.refreshable is of the following. I want to route to multiple pages using the same UI function.

My function is set up like this:
```python
@ui.refreshable
@router.page('/{user}/ui_page')
def container(user):
     data = data_store.get_user_data(user) #let's say we are retrieving some using data
     select = ui.select([1,2,3]).bind_value(data.select)
```
Let's say I create to two pages: /a/ui_page and /b/ui_page, if I'm going through some workflow while on /a/ui_page and call container.refresh(), the content in /b/ui_page will also be refreshed. However I'm trying to avoid this and only refresh the content in a's page.

My current workaround is creating user specific container instances, clear the container, and then repopulate the content. But I'm hoping there's a easier way to do this (hence looking for a way to reset elements in a container without repopulating).

Thanks.

**Reply by falkoschindler** [reply]
Hi @BillLi1992,
If I understand correctly, you try to have individual refreshables per client. Have a look at https://github.com/zauberzeug/nicegui/discussions/2477 for a very similar discussion.
